dask/array/linalg.py

```diff
@@ -1188,14 +1184,18 @@ def _key(i, j):
                         prevs.append(prev)
                     target = (operator.sub, target, (sum, prevs))
                 dsk[_key(i, j)] = (
-                    scipy.linalg.solve_triangular,
+                    solve_triangular_safe,
                     (a.name, i, i),
                     target,
                 )
 
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=[a, b])
+
+    a_meta = meta_from_array(a)
+    b_meta = meta_from_array(b)
     res = _solve_triangular_lower(
-        np.array([[1, 0], [1, 2]], dtype=a.dtype), np.array([0, 1], dtype=b.dtype)
+        np.array([[1, 0], [1, 2]], dtype=a.dtype, like=a_meta),
+        np.array([0, 1], dtype=b.dtype, like=b_meta),
```

Comment:
Perhaps we should use `array_safe` here for smoothly support the `like=` keyword?

https://github.com/dask/dask/blob/865515ca94b0da2872413d5d38b8ecf1e6b9fe1c/dask/array/utils.py#L467

cc @pentschev

Reply:
Yes!  I had meant to come back to that. Thanks for the reminder @jrbourbeau